hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/audit/TestS3AAuditLogMerger.java

```diff
@@ -0,0 +1,182 @@
+   */
+  @Test
+  public void testMergedFile() throws IOException {
+    //Testing the merged file with contents of first directory
+    dir1 = Files.createTempDirectory("sampleFilesDirectory").toFile();
+    file1 = File.createTempFile("sampleFile1", ".txt", dir1);
+    file2 = File.createTempFile("sampleFile2", ".txt", dir1);
+    try (FileWriter fw = new FileWriter(file1);
+        FileWriter fw1 = new FileWriter(file2)) {
+      fw.write("abcd");
+      fw1.write("efgh");
+    }
+    s3AAuditLogMerger.mergeFiles(dir1.getPath());
+    String str =
+        new String(Files.readAllBytes(Paths.get(auditLogFile.getPath())));
+    //File content of each audit log file in merged audit log file are
+    // divided by '\n'.
+    // Performing assertions will be easy by replacing '\n' with ''
+    String fileText = str.replace("\n", "");
+    assertTrue("the string 'abcd' should be in the merged file",
+        fileText.contains("abcd"));
+    assertTrue("the string 'efgh' should be in the merged file",
+        fileText.contains("efgh"));
+    assertFalse("the string 'ijkl' should not be in the merged file",
+        fileText.contains("ijkl"));
+
+    //Testing the merged file with contents of second directory
+    dir2 = Files.createTempDirectory("sampleFilesDirectory1").toFile();
+    file3 = File.createTempFile("sampleFile3", ".txt", dir2);
+    try (FileWriter fw2 = new FileWriter(file3)) {
+      fw2.write("ijkl");
+    }
+    s3AAuditLogMerger.mergeFiles(dir2.getPath());
+    String str1 =
+        new String(Files.readAllBytes(Paths.get(auditLogFile.getPath())));
+    //File content of each audit log file in merged audit log file are
+    // divided by '\n'.
+    // Performing assertions will be easy by replacing '\n' with ''
+    String fileText1 = str1.replace("\n", "");
+    assertFalse("the string 'abcd' should not be in the merged file",
+        fileText1.contains("abcd"));
+    assertFalse("the string 'efgh' should not be in the merged file",
+        fileText1.contains("efgh"));
+    assertTrue("the string 'ijkl' should be in the merged file",
+        fileText1.contains("ijkl"));
+  }
+
+  /**
+   * Testing the mergeFiles method in Merger class.
+   * by passing an empty directory and checks if merged file is created or not.
+   */
+  @Test
+  public void testMergeFilesEmptyDir() throws IOException {
+    dir1 = Files.createTempDirectory("emptyFilesDirectory").toFile();
+    if (auditLogFile.exists()) {
+      LOG.info("AuditLogFile already exists and we are deleting it here");
+      if (auditLogFile.delete()) {
+        LOG.debug("AuditLogFile deleted");
+      }
+    }
+    s3AAuditLogMerger.mergeFiles(dir1.getPath());
+    assertFalse(
+        "This AuditLogFile shouldn't exist if input directory is empty ",
+        auditLogFile.exists());
+  }
+
+  /**
+   * Delete all the sample directories and sample files after all tests.
+   */
+  @After
+  public void tearDown() throws Exception {
+    if (auditLogFile.exists()) {
```

Comment:
Use a base directory and delete all the children files in one go rather than one by one.

Reply:
actually used sample directories and files which will be deleted automatically